packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -212,6 +211,16 @@
       this.emitQueryPreviewRequestUpdated.cancel();
     }
 
+    /**
+     * Removes all the QueriesPreview from the state when the component is destroyed.
+     * This will prevent re-rendering effects when a query is selected and then deleted from the searchbox.
+     *
+     * @internal
+     */
+    protected destroyed(): void {
+      this.$x.emit('QueriesPreviewDestroyedHook');
+    }
```

Comment:
Emit `QueryPreviewUnmountedHook` when the component unmounts instead.

Reply:
Hi! I have tried that option and it ends up re-rendering the second carousel instead of the first. It only worked adequately removing all the list from the state. 
I have changed to emit the event in the QueriesPreviewList instead of the QueryPreview, as I think it makes more sense.